acvm-repo/bn254_blackbox_solver/src/lib.rs

```diff
@@ -47,11 +48,17 @@ impl BlackBoxFunctionSolver<FieldElement> for Bn254BlackBoxSolver {
     ) -> Result<(FieldElement, FieldElement), BlackBoxResolutionError> {
         let inputs: Vec<grumpkin::Fq> = inputs.iter().map(|input| input.into_repr()).collect();
         let result = pedersen::commitment::commit_native_with_index(&inputs, domain_separator);
-        let res_x =
-            FieldElement::from_repr(*result.x().expect("should not commit to point at infinity"));
-        let res_y =
-            FieldElement::from_repr(*result.y().expect("should not commit to point at infinity"));
-        Ok((res_x, res_y))
+        if result.is_zero() {
+            Ok((FieldElement::from(0_u128), FieldElement::from(0_u128)))
+        } else {
+            let res_x = FieldElement::from_repr(
+                *result.x().expect("should not commit to point at infinity"),
+            );
+            let res_y = FieldElement::from_repr(
+                *result.y().expect("should not commit to point at infinity"),
+            );
+            Ok((res_x, res_y))
+        }
```

Comment:
```suggestion
        let result = if let (x, y) = result.xy() {
            (FieldElement::from_repr(*x), FieldElement::from_repr(*y))
        } else {
            (FieldElement::zero(), FieldElement::zero())
        };
        
        Ok(result)
```

Reply:
Done (but not using the zero trait which is not in scope).